# Functions Should Not Have Too Many Parameters

## Rule Details

- **Rule ID:** go:S1763  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 20 minutes  
- **Type:** Code Smell  
- **Severity:** Major  
- **Tags:** Brain Overload, Clean Code  
- **Attribute:** Adaptability | Focused  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

Functions with a long parameter list are difficult to use, as maintainers must figure out the role of each parameter and keep track of their position.

## Noncompliant Code Example

```go
func setCoordinates(x1 int, y1 int, z1 int, x2 int, y2 int, z2 int) { 
    // Noncompliant
    // ...
}
```

## Compliant Solutions

### Split the Function into Smaller Ones

Each function should handle a distinct part of what the original function was doing, reducing confusion.

```go
func setOrigin(x int, y int, z int) { 
    // ...
}

func setSize(width int, height int, depth int) { 
    // ...
}
```

### Use a More Suitable Data Structure

Grouping related parameters into a structured type makes the function easier to use and understand.

```go
type point struct {
    x int
    y int
    z int
}

func setCoordinates(p1 point, p2 point) { 
    // ...
}
```

## Parameters

- **Max** - Maximum authorized number of parameters.  
  - **Default Value:** 7  

