# Functions Should Not Be Empty

## Rule Details

- **Rule ID:** go:S1186  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 5 minutes  
- **Type:** Code Smell  
- **Severity:** Critical  
- **Tags:** Suspicious, Clean Code  
- **Attribute:** Intentionality | Complete  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

There are several reasons why a function might be empty:

1. **Unintentional omission:** This should be fixed to prevent unexpected behavior in production.
2. **Not yet, or never will be, supported:** In this case, an exception should be thrown.
3. **Intentionally blank override:** A nested comment should explain the reason for the blank override.

## Noncompliant Code Example

```go
func doNothing() { // Noncompliant
}
```

## Compliant Solution

```go
func doNothing() {
    // Do nothing because of X and Y.
}
```

