# Track Uses of "TODO" Tags

## Rule Details

- **Rule ID:** go:S1135  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 0 minutes  
- **Type:** Code Smell  
- **Severity:** Info  
- **Tags:** CWE, Clean Code  
- **Attribute:** Intentionality | Complete  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

Developers often use `TODO` tags to mark areas in the code where **additional work or improvements** are needed but are not implemented immediately. However, these `TODO` tags sometimes get **overlooked or forgotten**, leading to **incomplete or unfinished code**.

This rule **identifies and addresses unattended `TODO` tags** to ensure a **clean and maintainable codebase**.

## What is the Potential Impact?

Unattended `TODO` tags in code can have **significant implications**:

- **Incomplete Functionality:** Unimplemented `TODO` tags result in missing functionality, leading to unexpected behavior and a poor end-user experience.
- **Missed Bug Fixes:** Unaddressed `TODO` tags may cause **critical bug fixes and security updates** to be overlooked.
- **Impact on Collaboration:** In team environments, unattended `TODO` tags can create confusion, leading to **conflicts and redundant efforts**.
- **Codebase Bloat:** Accumulation of unattended `TODO` tags can clutter the codebase, making it harder to **distinguish between work in progress and completed code**.

Addressing this code smell is **essential** for maintainability, readability, and effective collaboration among developers.

## Noncompliant Code Example

```go
func foo() {
    // TODO
}
```

