# Functions Should Not Have Identical Implementations

## Rule Details

- **Rule ID:** go:S4144  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 15 minutes  
- **Type:** Code Smell  
- **Severity:** Major  
- **Tags:** Confusing, Clean Code  
- **Attribute:** Adaptability | Distinct  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

Two functions having the same implementation are **suspicious**. It might indicate:

- A copy-paste error where something else was intended.
- An intentional duplication, which becomes a maintenance burden.

## Noncompliant Code Example

```go
func fun1() (x, y int) {
    a, b := 1, 2
    b, a = a, b
    return a, b
}

func fun2() (x, y int) { // Noncompliant; duplicates fun1
    a, b := 1, 2
    b, a = a, b
    return a, b
}
```

## Compliant Solution

```go
func fun1() (x, y int) {
    a, b := 1, 2
    b, a = a, b
    return a, b
}

func fun2() (x, y int) { // Intent is clear
    return fun1()
}
```

## Exceptions

- Functions with fewer than **2 statements** are ignored.

