# String Literals Should Not Be Duplicated

## Rule Details

- **Rule ID:** go:S1192  
- **Analysis Scope:** Main sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 2 minutes  
- **Type:** Code Smell  
- **Severity:** Critical  
- **Tags:** Design, Clean Code  
- **Attribute:** Adaptability | Distinct  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

Duplicated **string literals** make refactoring **error-prone**, as all occurrences must be updated manually. Using **constants** allows a single reference to be updated in one place, improving maintainability and reducing errors.

## Noncompliant Code Example

With the default threshold of **3**:

```go
func run() {
    prepare("This should be a constant") // Noncompliant; 'This should ...' is duplicated 3 times
    execute("This should be a constant")
    release("This should be a constant")
}
```

## Compliant Solution

```go
const ACTION = "This should be a constant"

func run() {
    prepare(ACTION)
    execute(ACTION)
    release(ACTION)
}
```

## Exceptions

To prevent false positives, this rule ignores:

- **Literals with 5 or fewer characters**
- **Literals containing only letters, digits, and `_`**

## Parameters

- **threshold** - Number of times a literal must be duplicated to trigger an issue.  
  - **Default Value:** 3  

