# Control Flow Statements "if", "for", and "switch" Should Not Be Nested Too Deeply

## Rule Details

- **Rule ID:** go:S134  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 10 minutes  
- **Type:** Code Smell  
- **Severity:** Critical  
- **Tags:** Brain Overload, Clean Code  
- **Attribute:** Adaptability | Focused  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

Nested `if`, `for`, `while`, `switch`, and `try` statements are key contributors to what is known as **"Spaghetti code"**. Such code is difficult to read, refactor, and maintain. Reducing nesting depth improves code clarity and maintainability.

## Noncompliant Code Example

With the default threshold of **3**:

```go
if condition1 { // Compliant - depth = 1
    /* ... */
    if condition2 { // Compliant - depth = 2
        /* ... */
        for i := 1; i <= 10; i++ { // Compliant - depth = 3, not exceeding the limit
            /* ... */
            if condition4 { // Noncompliant - depth = 4
                if condition5 { // Depth = 5, exceeding the limit, but issues are only reported on depth = 4
                    /* ... */
                }
                return
            }
        }
    }
}
```

## Parameters

- **max** - Maximum allowed control flow statement nesting depth.  
  - **Default Value:** 4  

