# Statements Should Be on Separate Lines

## Rule Details

- **Rule ID:** go:S122  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 1 minute  
- **Type:** Code Smell  
- **Severity:** Major  
- **Tags:** Convention, Clean Code  
- **Attribute:** Consistency | Formatted  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

Placing multiple statements on a single line reduces **code readability** and makes **debugging more complex**.

## Noncompliant Code Example

```go
foo(); bar() // Noncompliant
```

## Compliant Solution

Write one statement per line to improve readability:

```go
foo()
bar()
```

## Exceptions

This rule **ignores** control flow statements with a **single nested statement**.

```go
if condition {
    doSomething() // Compliant by exception
}
```

