# Identical Expressions Should Not Be Used on Both Sides of a Binary Operator

## Rule Details

- **Rule ID:** go:S1764  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 2 minutes  
- **Type:** Bug  
- **Severity:** Major  
- **Tags:** None  
- **Attribute:** Intentionality | Logical  
- **Software Qualities Impacted:** Reliability  

## Why is this an Issue?

Using the same value on both sides of a binary operator is a **code defect**. 

- For **logical operators**, it is either a **copy/paste error** (a bug) or **duplicated code** that should be simplified.
- For **bitwise operators** and most **binary mathematical operators**, having the same value on both sides produces predictable results and should be simplified.

## Noncompliant Code Example

```go
func main() {
    v1 := (true && false) && (true && false) // Noncompliant
}
```

## Compliant Solution

```go
func main() {
    v1 := (true && false) // Compliant
}
```

## Exceptions

This rule ignores the following operators:

- `*`
- `+`
- `<<`
- `=`

