# Expressions Should Not Be Too Complex

## Rule Details

- **Rule ID:** go:S1067  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 5 minutes  
- **Type:** Code Smell  
- **Severity:** Critical  
- **Tags:** Brain Overload, Clean Code  
- **Attribute:** Adaptability | Focused  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

The complexity of an expression is defined by the number of `&&`, `||`, and ternary (`condition ? ifTrue : ifFalse`) operators it contains. A single expressionâ€™s complexity should not become too high to maintain readability and simplify debugging.

## Noncompliant Code Example

With the default threshold value of **3**:

```go
if (((condition1 && condition2) || (condition3 && condition4)) && condition5) {
    // Noncompliant
}
```

## Compliant Solution

```go
if ((myFirstCondition() || mySecondCondition()) && myLastCondition()) {
    // Compliant
}
```

## Parameters

- **max** - Maximum number of allowed conditional operators in an expression.  
  - **Default Value:** 3  

