
## Rule Details

- **Rule ID:** go:S1151  
- **Analysis Scope:** Main sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 5 minutes  
- **Type:** Code Smell  
- **Severity:** Major  
- **Tags:** Brain Overload, Clean Code  
- **Attribute:** Adaptability | Focused  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

The `switch` statement should be used only to clearly define distinct branches in the control flow. When a `case` clause contains too many statements, it significantly decreases the readability of the overall control flow statement.

In such cases, the content of the `case` clause should be extracted into a dedicated method to improve maintainability and readability.

## Noncompliant Code Example

With the default threshold of 5 lines:

```go
func foo(tag int) {
    switch tag {
    case 0:
        methodCall1()
        methodCall2()
        methodCall3()
        methodCall4()
        methodCall5()
        methodCall6()
    case 1:
        bar()
    }
}
```

## Compliant Solution

```go
func foo(tag int) {
    switch tag {
    case 0:
        executeAll()
    case 1:
        bar()
    }
}

func executeAll() {
    methodCall1()
    methodCall2()
    methodCall3()
    methodCall4()
    methodCall5()
    methodCall6()
}
```

## Parameters

- **max** - Maximum number of lines in a `case` clause.  
  - **Default Value:** 6  

