
## Rule Details

- **Rule ID:** go:S1821  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 10 minutes  
- **Type:** Code Smell  
- **Severity:** Critical  
- **Tags:** Pitfall, Clean Code  
- **Attribute:** Adaptability | Focused  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

Nested `switch` structures are difficult to understand because cases of an inner `switch` can easily be confused as belonging to an outer statement. Therefore, nested `switch` statements should be avoided.

Specifically, you should structure your code to avoid the need for nested `switch` statements. If this is not possible, consider moving the inner `switch` to another function.

## Noncompliant Code Example

```go
func foo(x, y int) {
    switch x {
    case 0:
        switch y { // Noncompliant; nested switch
            // ...
        }
    case 1:
        // ...
    default:
        // ...
    }
}
```

## Compliant Solution

```go
func foo(x, y int) {
    switch x {
    case 0:
        bar(y)
    case 1:
        // ...
    default:
        // ...
    }
}

func bar(y int) {
    switch y {
        // ...
    }
}
```

