
## Rule Details

- **Rule ID:** go:S126  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 5 minutes  
- **Type:** Code Smell  
- **Severity:** Critical  
- **Tags:** Clean Code  
- **Attribute:** Intentionality | Clear  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

This rule applies whenever an `if` statement is followed by one or more `else if` statements; the final `else if` should be followed by an `else` statement. 

The requirement for a final `else` statement is **defensive programming**. The `else` statement should either take appropriate action or contain a suitable comment explaining why no action is needed. This is consistent with the requirement to have a final `default` clause in a `switch` statement.

## Noncompliant Code Example

```go
if x == 0 {
    doSomething()
} else if x == 1 {
    doSomethingElse()
}
```

## Compliant Solution

```go
if x == 0 {
    doSomething()
} else if x == 1 {
    doSomethingElse()
} else {
    return errors.New("unsupported int")
}
```

## Exceptions

When all branches of an `if - else if` construct end with `return` or `break`, the code that follows the `if` implicitly behaves as if it was inside an `else` clause. This rule will therefore ignore such cases.

