# Variables Should Not Be Self-Assigned

## Rule Details

- **Rule ID:** go:S1656  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 3 minutes  
- **Type:** Bug  
- **Severity:** Major  
- **Tags:** None  
- **Attribute:** Intentionality | Logical  
- **Software Qualities Impacted:** Reliability  

## Why is this an Issue?

Re-assigning a variable to itself **serves no purpose**. It is either:

- **Redundant** and should be removed.
- **A mistake**, where another value or variable was **intended** for assignment.

## Noncompliant Code Example

```go
func (user *User) rename(name string) {
    name = name // Noncompliant
}
```

## Compliant Solution

```go
func (user *User) rename(name string) {
    user.name = name // Compliant
}
```

