# Nested Blocks of Code Should Not Be Left Empty

## Rule Details

- **Rule ID:** go:S108  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 5 minutes  
- **Type:** Code Smell  
- **Severity:** Major  
- **Tags:** Suspicious, Clean Code  
- **Attribute:** Intentionality | Clear  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

An **empty code block** is confusing and requires additional effort from maintainers to determine if it was **intentional** or if the implementation is **incomplete**.

### Noncompliant Code Example

```go
func compute(a int, b int) {
    sum := a + b
    if sum > 0 { 
        // Noncompliant; empty on purpose or missing implementation?
    }
    fmt.Println("Result:", sum)
}
```

Removing or filling the empty code blocks removes ambiguity and results in clearer, more maintainable code.

## Exceptions

This rule ignores:

- Code blocks that contain **comments**
- `for` loops **without** init and post statements that have empty blocks

