# Boolean Literals Should Not Be Redundant

## Rule Details

- **Rule ID:** go:S1125  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 5 minutes  
- **Type:** Code Smell  
- **Severity:** Minor  
- **Tags:** Clumsy, Clean Code  
- **Attribute:** Consistency | Conventional  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

Redundant Boolean literals should be removed from expressions to improve readability and maintainability. Explicitly including `true` or `false` in logical expressions adds unnecessary complexity and should be avoided.

## Noncompliant Code Example

```go
if boolFunc() || false { 
    // ... 
}

flag := x && true
```

## Compliant Solution

```go
if boolFunc() { 
    // ... 
}

flag := x
```

