# All Code Should Be Reachable

## Rule Details

- **Rule ID:** go:S1763  
- **Analysis Scope:** Main sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 5 minutes  
- **Type:** Bug  
- **Severity:** Major  
- **Tags:** CWE, Clean Code  
- **Attribute:** Intentionality | Logical  
- **Software Qualities Impacted:** Reliability  

## Why is this an Issue?

Once control flow has been moved out of the current code block, any subsequent statements become effectively unreachable. Some statements such as `return`, `break`, `continue`, `goto`, `switch`, and `throw` expressions move control flow out of the current code block. Any unlabeled statements that follow such a jump are considered **dead code** and should either be removed or corrected.

## Noncompliant Code Example

```go
func add(x, y int) int {
    return x + y // Noncompliant
    z := x + y  // Dead code
}
```

## Compliant Solution

```go
func add(x, y int) int {
    return x + y // Compliant
}
```

