# Octal Values Should Not Be Used

## Rule Details

- **Rule ID:** go:S1314  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 5 minutes  
- **Type:** Code Smell  
- **Severity:** Blocker  
- **Tags:** Pitfall, Clean Code  
- **Attribute:** Intentionality | Clear  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

Integer literals starting with a **zero (0)** are interpreted as **octal values** rather than decimal values. While octal values are fully supported, most developers **lack experience** with them and may **mistakenly** assume they are decimal values.

## Noncompliant Code Example

```go
func printTen() {
    myNumber := 010 // Noncompliant. myNumber will hold 8, not 10 - was this really expected?
    fmt.Println(myNumber)
}
```

## Compliant Solution

```go
func printTen() {
    myNumber := 10 // Compliant
    fmt.Println(myNumber)
}
```

