# Useless `if(true) {...}` and `if(false){...}` Blocks Should Be Removed

## Rule Details

- **Rule ID:** go:S1145  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 2 minutes  
- **Type:** Bug  
- **Severity:** Major  
- **Tags:** CWE, Clean Code  
- **Attribute:** Consistency | Conventional  
- **Software Qualities Impacted:** Reliability  

## Why is this an Issue?

`if` statements with conditions that are **always false** make blocks of code **non-functional**. 

`if` statements with conditions that are **always true** are **completely redundant**, making the code **less readable**.

### Possible Causes:
1. An `if` statement was changed during debugging and committed mistakenly.
2. Some value was left **unset**.
3. Some logic **is not behaving as expected**.

In any of these cases, **unconditional `if` statements should be removed**.

## Noncompliant Code Example

```go
if true {
    doSomething()
}

if false {
    doSomething()
}
```

## Compliant Solution

```go
doSomething()
```

