# Track Uses of "FIXME" Tags

## Rule Details

- **Rule ID:** go:S1134  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 0 minutes  
- **Type:** Code Smell  
- **Severity:** Major  
- **Tags:** CWE, Clean Code  
- **Attribute:** Intentionality | Complete  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

`FIXME` tags are commonly used to **mark places where a bug is suspected** but is deferred for later fixing. 

Developers may:
- **Not have time** to address the issue immediately.
- **Forget** about the tag, leading to unresolved bugs in production.

This rule tracks `FIXME` tags to **ensure they do not go unnoticed**.

## Noncompliant Code Example

```go
func foo() {
    // FIXME
}
```

