
## Rule Details

- **Rule ID:** go:S131  
- **Analysis Scope:** All sources  
- **Rule Repository:** Sonar (Go)  
- **Effort:** 5 minutes  
- **Type:** Code Smell  
- **Severity:** Critical  
- **Tags:** CWE, Clean Code  
- **Attribute:** Intentionality | Clear  
- **Software Qualities Impacted:** Maintainability  

## Why is this an Issue?

The requirement for a final `default` clause is **defensive programming**. The `default` clause should either take appropriate action or contain a suitable comment explaining why no action is needed. 

## Noncompliant Code Example

```go
switch tag {
    // Noncompliant - default case is missing
    case 0, 1, 2, 3:
        foo()
    case 4, 5, 6, 7:
        bar()
}
```

## Compliant Solution

```go
switch tag {
    case 0, 1, 2, 3:
        foo()
    case 4, 5, 6, 7:
        bar()
    default:
        qix()
}
```

